kubernetes-server-mock/src/main/java/io/fabric8/kubernetes/client/server/mock/KubernetesCrudDispatcher.java

```diff
@@ -66,7 +66,7 @@ public KubernetesCrudDispatcher(KubernetesCrudAttributesExtractor attributeExtra
   }
 
   @Override
-  public MockResponse dispatch(RecordedRequest request) {
+  public synchronized MockResponse dispatch(RecordedRequest request) {
```

Comment:
A better alternative would maybe be to make the `map` thread safe in `io.fabric8.mockwebserver.crud.CrudDispatcher`, but that would require API change in `mockwebserver`

Reply:
Given how the people tend to use the mockserver, it probably makes sense that the method is blocking and synchronized (however, this is probably encouraging bad-practices).